Automatically load terrain textures.

Shadowfita opened the pull request:
Unsure if this is the right way to go about it. Tilesize may be incorrect. Not 100% done.

Shadowfita committed:
Automatically load terrain textures.

Unsure if this is the right way to go about it. Tilesize may be incorrect. Not 100% done.

Ben1138 commented on SWBF2Import/SWBF2Import.cs:
```diff
@@ -138,7 +138,19 @@ public static class SWBF2Import {
             data.size = new Vector3(size, range, size);
 
+			//Load textures onto terrain
+            SplatPrototype[] splats = new SplatPrototype[terrain.Layers.Where(l => !string.IsNullOrEmpty(parseChars(l.DiffuseTexture))).ToArray().Length];
```
You can use .Count() instead of .ToArray().Length

Shadowfita replied:
Whoops! Didn't think about that. Haha.

Ben1138 commented on SWBF2Import/SWBF2Import.cs:
```diff
@@ -151,6 +163,11 @@ public static class SWBF2Import {
         }
     }
 
+    public static string parseChars(String str)
```
Maybe rename the method to sth that descibes what it does. Is public necessary?

Shadowfita replied:
Public is unnecessary. It's possible the method is not even needed and I was just making assumptions but the layer.DiffuseTexture strings have got a lot of invisible characters and I was having issues with some things then added that and may not need it anymore. Though it is probably needed for my string.NullOrEmpty check since the "invisible chars" dont count as whitespace and trim() doesnt change anything. "parseChars" uses regex to filter out the invisible characters that don't count as whitespace.

Ben1138 replied:
I see. IIRC grabbing the layer texture strings just performs a copy of the respective string block in the .ter file without any checks, which might contain empty bytes (00).

I wont though patch this in the C# version, but consider it in the C++ version (which should replace the C# version in the long term).

Ben1138 reviewed (commented):
Looks good, haven't tested it yet. I'm looking forward to see more :)

Shadowfita committed:
Shader change and prefab creation

-Updated Shader name for latest unity version
-Automatically creates prefabs for BF2 MSH and saves them into their respective "world" folder

The current changes require all the BF2 .tga textures to be in "Assets\Resources\Textures" for the way I've got prefabs running at the moment. It's a bit hacky currently, will improve later.
